Design/SANS/BeamCentreFinderDesign.md

```diff
@@ -0,0 +1,271 @@
+# Design Document for Beam Centre Finder
+
+### Current issue
+In order to calibrate the Beam Centre position with respect to the detectors, an accurate estimation estimation of the position needs to be performed. This functionality is provided by a Beam Centre Finder (*BCF*). 
+
+An implementation for this BCF already exists for ISIS SANS. This implementation is heavily tied up with the ISIS instruments and the specific way the data reduction is realized for ISIS SANS. ANSTO requires the same functionality as the current *BCF* provides. Unfortunately, it is not possible to directly use the *BCF*'s functionality with the ANSTO specific implementation of data reduction.
+
+The different approaches of ISIS and ANSTO are currently not compatible. They can be summarized as:
+
+* ISIS makes use of a so-called ReductionSingleton which has a series of registered reduction steps. During the reduction, the ReductionSingleton() carries state (mainly via the custom ISIS instrument classes). Intermediate results are stored in the Analysis Data Service (ADS).
```

Comment:
Might be worth specifying more about the ISIS Instrument. i.e. what it's responsibilities are.